TAO/tests/CORBA_is_nil/client.cpp

```diff
@@ -0,0 +1,96 @@
+#include "TestC.h"
+#include "ace/Get_Opt.h"
+
+const ACE_TCHAR *ior = ACE_TEXT ("file://test.ior");
+
+int process_result = 1;
+
+int parse_args (int argc, ACE_TCHAR *argv[])
+{
+ACE_Get_Opt get_opts (argc, argv, ACE_TEXT("f:"));
+  int c;
+
+  while ((c = get_opts ()) != -1)
+    switch (c)
+      {
+      case 'f':
+        ior = get_opts.opt_arg ();
+        break;
+
+      case '?':
+      default:
+        ACE_ERROR_RETURN ((LM_ERROR,
+                           "usage:  %s "
+                           "-f <ior> "
+                           "\n",
+                           argv [0]),
+                          -1);
+      }
+  // Indicates successful parsing of the command line
+  return 0;
+}
+
+int
+ACE_TMAIN(int argc, ACE_TCHAR *argv[])
+{
+  try
+    {
+      CORBA::ORB_var orb = CORBA::ORB_init (argc, argv);
+
+      if (parse_args (argc, argv) != 0)
+        return 1;
+
+      CORBA::Object_var tmp = orb->string_to_object(ior);
+
+      Test::Hello_var hello = Test::Hello::_narrow(tmp.in ());
+
+      if (CORBA::is_nil (hello.in ()))
+      {
+         ACE_ERROR_RETURN ((LM_DEBUG,
+                            "Nil Test::Hello reference <%s>\n",
+                            ior),
+                           1);
+      }
+
+      CORBA::String_var the_string = hello->get_string ();
+
+      ACE_DEBUG ((LM_DEBUG,
+                  "Hello from file IOR: %s\n",
+                  the_string.in()));
+
+      Test::Hello_var helloRef = hello->get_Hello();
+
+      if (CORBA::is_nil(helloRef.in()))
+      {
+         ACE_DEBUG ((LM_DEBUG,
+                     ACE_TEXT("Correctly detected nil Object reference.\n")));
+         process_result = 0;
+      }
+      else
+      {
+         CORBA::Object_ptr obj = helloRef.in();
```

Comment:
Can you add a duplicate to a CORBA::Object which you assign into a _var to make sure that that also works correctly?

Reply:
Added a call to _duplicate to confirm that the nil reference is detected. This was
clear from inspection because _narrow changes the reference to a nullptr, but
useful to ensure that future changes don't break this.